lib/routes/users-request-reset.js

```diff
@@ -27,20 +28,28 @@ module.exports = (server, options) => {
         handler: async (request) => {
 
             const Users = request.models().Users;
+            const { emailService } = request.services();
             const Payload = request.payload;
 
             const rawResetToken = Buffer.from(Uuid({ rng: Uuid.nodeRNG }));
 
             const hash = Pwd.hashSync(rawResetToken);
 
-            await Users.query()
+            const patchedUser = await Users.query()
                 .patch({ 'password': null, resetToken: hash.toString('utf8') })
-                .where({ 'email': Payload.email });
-            //send token to user via email here. This functionality
-            //intentionally left out, as it is dependent on
-            //infrastructure and front end implementation.
+                .where({ 'email': Payload.email })
+                .first()
+                .returning('*');
 
-            return 'Raw Reset Token, for testing only *' + rawResetToken + '* TODO: Implement email sending';
+            if (patchedUser) {
+
+                const resetUrl = options.siteUrl + '/password-reset/' + rawResetToken;
+
+                await emailService.send('password-reset', patchedUser, { resetUrl });
```

Comment:
If you want email-sending to be totally centralized then you could emit a server event here, and the email service could listen for it.  This honestly looks great to me though, just ideas :)